common.c

```diff
@@ -52,8 +57,22 @@
  * calls into your tapped version of getuid and we
  * get into an infinite loop.
  *
+ * g_enable_tracing is used only for the main thread,
+ * if we see ourselves being called from a different thread
+ * we will initialized the key g_enable_tracing_key
+ * and will use that to store a per-thread tracing
+ * enabled/disabled setting. I used to declare
+ * g_enable_tracing with the __thread modifier, but in
+ * macOS that will eventually call malloc which would
+ * get us into a loop.
+ *
+ * g_enable_tracing_key logic is backwards, 0 (NULL)
+ * means tracing is enabled. 1 means tracing is disabled.
+ *
```

Comment:
I can't call malloc here, would basically break everything since we haven't disabled traced yet (we are trying to). Passing an int cast to a void * to pthread_setspecific is perfectly acceptable here. If will remove the backward logic and change it to something more clear.